airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -1600,6 +1602,74 @@ public boolean getConnectionHasAlphaOrBetaConnector(final UUID connectionId) thr
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .fetch())
+        .map(DbConverter::buildConnectorBuilderProject)
+        .stream()
+        .findFirst();
+  }
+
+  public Stream<ConnectorBuilderProject> getConnectorBuilderProjectsByWorkspace(final UUID workspaceId) throws IOException {
+    final Condition matchByIds = CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID.eq(workspaceId);
+
+    return database
+        .query(ctx -> ctx
+            .select(CONNECTOR_BUILDER_PROJECT.ID, CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, CONNECTOR_BUILDER_PROJECT.NAME,
+                CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID)
+            .from(CONNECTOR_BUILDER_PROJECT)
+            .where(matchByIds)
+            .fetch())
+        .map(DbConverter::buildConnectorBuilderProjectWithoutManifestDraft)
+        .stream();
+  }
+
+  public boolean deleteBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.transaction(ctx -> ctx.deleteFrom(CONNECTOR_BUILDER_PROJECT))
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .execute() > 0;
+  }
+
+  public void writeBuilderProject(final ConnectorBuilderProject builderProject) throws IOException {
+    database.transaction(ctx -> {
+      final OffsetDateTime timestamp = OffsetDateTime.now();
+      final Condition matchId = CONNECTOR_BUILDER_PROJECT.ID.eq(builderProject.getBuilderProjectId());
+      final boolean isExistingConfig = ctx.fetchExists(select()
+          .from(CONNECTOR_BUILDER_PROJECT)
+          .where(matchId));
+
+      if (isExistingConfig) {
+        ctx.update(CONNECTOR_BUILDER_PROJECT)
+            .set(CONNECTOR_BUILDER_PROJECT.ID, builderProject.getBuilderProjectId())
+            .set(CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, builderProject.getWorkspaceId())
+            .set(CONNECTOR_BUILDER_PROJECT.NAME, builderProject.getName())
+            .set(CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID, builderProject.getActorDefinitionId())
+            .set(CONNECTOR_BUILDER_PROJECT.MANIFEST_DRAFT, JSONB.valueOf(Jsons.serialize(builderProject.getManifestDraft())))
+            .set(WORKSPACE.UPDATED_AT, timestamp)
+            .where(matchId)
+            .execute();
+      } else {
+        ctx.insertInto(CONNECTOR_BUILDER_PROJECT)
+            .set(CONNECTOR_BUILDER_PROJECT.ID, builderProject.getBuilderProjectId())
+            .set(CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, builderProject.getWorkspaceId())
+            .set(CONNECTOR_BUILDER_PROJECT.NAME, builderProject.getName())
+            .set(CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID, builderProject.getActorDefinitionId())
+            .set(CONNECTOR_BUILDER_PROJECT.MANIFEST_DRAFT, JSONB.valueOf(Jsons.serialize(builderProject.getManifestDraft())))
+            .set(WORKSPACE.CREATED_AT, timestamp)
+            .set(WORKSPACE.UPDATED_AT, timestamp)
+            .execute();
+      }
+      return null;
```

Comment:
Why the return statement here?

Reply:
It's a type-inference thing - without the explicit `null` return it doesn't have a type param for the generic type of the wrapping db context helper.